### PR TITLE
Behaviour tests base: context & steps library, sample feature test

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
## Usage and product changes

We introduce the `//tests/behaviour` package containing the `steps` library, which encompasses the BDD context and step implementations, and a sample BDD test for the database feature.

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -55,13 +55,13 @@ checkstyle_test(
         "LICENSE",
         "VERSION",
     ],
-    license_type = "agpl-header",
+    license_type = "mpl-header",
```
Comment (dmitrii-ubskii) [inline BUILD:58]: `agpl-header` is no longer legal with the updated dependencies

File: tests/behaviour/steps/connection/database.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        connection_create_database(context, name.into()).await;
+    }
+}
+
+#[apply(generic_step)]
+#[step(expr = "connection create databases in parallel:")]
+async fn connection_create_databases_in_parallel(context: &mut Context, step: &Step) {
+    todo!()
+    // join_all(util::iter_table(step).map(|name| create_database(&context.databases, name.to_string()))).await;
+}
```
Comment (dmitrii-ubskii) [inline tests/behaviour/steps/connection/database.rs:42]: Database creation currently requires `&mut Server`. Parallel operations are therefore left unimplemented.

File: tests/behaviour/steps/connection/database.rs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+#[apply(generic_step)]
+#[step(expr = "connection delete database: {word}")]
+pub async fn connection_delete_database(context: &mut Context, name: String) {
+    todo!()
+    // context.databases.get(name).and_then(Database::delete).await.unwrap();
+}
```
Comment (dmitrii-ubskii) [inline tests/behaviour/steps/connection/database.rs:49]: Database deletion is not implemented.

File: tests/behaviour/steps/lib.rs
Comment (dmitrii-ubskii) [inline tests/behaviour/steps/lib.rs:1]: Base behaviour test lib: context + step implementations. Step implementations cannot be split out to different libraries.

File: tests/behaviour/connection/database/database.rs
Comment (dmitrii-ubskii) [inline tests/behaviour/connection/database/database.rs:1]: Sample behaviour test.

File: database/BUILD [hunk 4]
```diff
@@ -43,5 +43,5 @@ checkstyle_test(
     exclude = glob([
         "Cargo.*",
     ]),
-    license_type = "agpl-header",
+    license_type = "mpl-header",
```
Comment (flyingsilverfin) [inline database/BUILD:46]: If you're gonna do this might as well update the copyright headers too!

File: server/typedb.rs [hunk 5]
```diff
@@ -59,10 +60,19 @@ impl Server {
...
+            .entry(name.to_owned())
+            .or_insert_with(|| Database::recover(&self.data_directory.join(name), name).unwrap());
+    }
+
+    pub fn databases(&self) -> &HashMap<String, Database<WAL>> {
+        &self.databases
+    }
+
+    pub fn serve(mut self) {
+        self.create_database("test");
```
Comment (flyingsilverfin) [inline server/typedb.rs:75]: Not sure we should create this DB in the serve!

**Pull request merged**